File: src/test/java/org/vanautrui/languages/parsing/astnodes/terminal/ExpressionNodeTest.java
package org.vanautrui.languages.parsing.astnodes.terminal;

import org.junit.Test;
import org.vanautrui.languages.lexing.utils.TokenList;
import org.vanautrui.languages.lexing.tokens.*;
import org.vanautrui.languages.parsing.astnodes.nonterminal.ExpressionNode;

public class ExpressionNodeTest {

    @Test
    public void test_simple_expression() throws Exception {

        TokenList list = new TokenList();
        list.add(new IntegerNonNegativeConstantToken(4));
        ExpressionNode expr = new ExpressionNode(list);
    }

    @Test
    public void test_variable_name_expression() throws Exception {

        TokenList list = new TokenList();
        list.add(new IdentifierToken("x"));
        ExpressionNode expr = new ExpressionNode(list);
    }


}

File: dragon-parser/src/test/astnodes/terminal/ExpressionNodeTest.cpp
package org.vanautrui.languages.test.astnodes.terminal;

import org.vanautrui.languages.compiler.lexing.tokens.IdentifierToken;
import org.vanautrui.languages.compiler.lexing.tokens.IntegerNonNegativeConstantToken;
import org.vanautrui.languages.compiler.lexing.tokens.StringConstantToken;
import org.vanautrui.languages.compiler.lexing.utils.TokenList;
import org.vanautrui.languages.compiler.parsing.astnodes.nonterminal.ExpressionNode;

import java.nio.file.Paths;

public final class ExpressionNodeTest {

	public static int test_simple_expression() {

		try {
			final TokenList list = new TokenList(Paths.get("/dev/null"));
			list.add(new IntegerNonNegativeConstantToken(4));
			final ExpressionNode expr = new ExpressionNode(list);

			return 1;
		}catch (Exception e){
			return 0;
		}
	}

	public static int test_variable_name_expression() {

		try {
			final TokenList list = new TokenList(Paths.get("/dev/null"));

			list.add(new IdentifierToken("x"));

			final ExpressionNode expr = new ExpressionNode(list);

			return 1;
		}catch (Exception e){
			return 0;
		}
	}

	public static int recognize_string_constant_expression() {

		try {
			final TokenList tokens = new TokenList(Paths.get("/dev/null"));

			tokens.add(new StringConstantToken("hello"));

			final ExpressionNode expr = new ExpressionNode(tokens);

			return 1;
		}catch (Exception e){
			return 0;
		}
	}
}
